src/stores/search.ts

```diff
@@ -156,11 +167,47 @@ export const useSearchStore = defineStore('search', {
       this.clearOtherMediaTypeFilters(type)
     },
     setSearchTerm(term: string) {
-      this.searchTerm = term.trim()
+      const formattedTerm = term.trim()
+      this.searchTerm = formattedTerm
+      this.addRecentSearch(formattedTerm)
+    },
+    /**
+     * This method need not exist and is only used to fix an odd
+     * hydration bug in the search route. After navigating from
+     * the homepage, the watcher in useStorage doesn't work.
+     */
+    refreshRecentSearches() {
+      // @ts-expect-error https://github.com/microsoft/TypeScript/issues/43826
+      this.recentSearches = useStorage<string[]>('recent-searches', [])
+    },
+    /** Add a new term to the list of recent search terms */
+    addRecentSearch(
+      search: string /** A search term to add to the saved list.*/
+    ) {
+      // No-op if feature isn't enabled
+      const featureFlagStore = useFeatureFlagStore()
+      const saveRecentSearches = featureFlagStore.isOn('recent_searches')
+      if (!saveRecentSearches) {
+        return
+      }
+
+      /**
+       * Add the latest search to the top of the stack,
+       * then add the existing items, making sure not to exceed
+       * the max count, and removing existing occurrences of the
+       * most latest search term, if there are any.
+       */
+      this.recentSearches = [
+        search,
+        ...this.recentSearches.filter((i) => i !== search),
+      ].slice(0, parseInt(env.savedSearchCount))
```

Comment:
Is it possible for `env.savedSearchCount` to be undefined? If it is, we should handle that possibility.

```suggestion
      ].slice(0, parseInt(env.savedSearchCount ?? '4'))
```